spec/grape/validations/params_scope_spec.rb

```diff
@@ -497,6 +497,34 @@ def initialize(value)
       expect(body.keys).to_not include('b')
     end
 
+    it 'allows aliasing of dependent on parameter' do
+      subject.params do
+        optional :a, as: :b
+        given b: ->(val) { val == 'x' } do
+          requires :c
+        end
+      end
+      subject.get('/') { declared(params) }
+
+      get '/', a: 'x'
+      expect(last_response.status).to eq 400
+      expect(last_response.body).to eq 'c is missing'
+
+      get '/', a: 'y'
+      expect(last_response.status).to eq 200
+    end
+
+    it 'raises an error if the dependent parameter is not the aliased one' do
+      expect do
+        subject.params do
+          optional :a, as: :b
+          given :a do
```

Comment:
I add a spec to show with this fix, `given :a`(not the aliased param `:b`) case.